src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -29,15 +30,17 @@ public Connection(ListenerContext context, UvStreamHandle socket) : base(context
             ConnectionControl = this;
 
             _connectionId = Interlocked.Increment(ref _lastConnectionId);
+
+            SocketInput = new SocketInput(Memory2);
+            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
+            _frame = new Frame(this);
         }
 
         public void Start()
         {
             Log.ConnectionStart(_connectionId);
 
-            SocketInput = new SocketInput(Memory);
-            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
-            _frame = new Frame(this);
+            Task.Run((Action)_frame.ProcessFraming);
```

Comment:
The timing of ReadStart doesn't matter in this case. It initiates the first read with libuv, but because this function is running on the uv loop it won't fire any callbacks until it returns.

Even if it did, it would still be safe because the read callback and the frame consumption are mediated by the SocketInput class.